### PR TITLE
Improve entry points for zip deploy

### LINKED ISSUE #15: Fix entry points for zip deploy
My first iteration of zip deploy has two entry points (command palette and functions explorer) with the same "Deploy as Zip Package" label.

Ideally we should have three entry points with a context-specific label for each. Initial proposal:
1. Command palette "Deploy as zip package"
1. Functions explorer: "Deploy here as zip package"
1. File explorer: "Deploy to Function App as zip package"

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
1. Add 'right click on folder' entry point
1. Add context-specific wording for each entry point
1. Make folder selection consistent across several commands
1. Refactor util.ts into multiple files

Fixes #15

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -144,10 +154,27 @@
...
+                {
+                    "command": "azureFunctions.deployZipFromFolder",
+                    "when": "explorerResourceIsFolder == true",
+                    "group": "zzz_@1"
+                }
+            ],
+            "commandPalette": [
+                {
+                    "command": "azureFunctions.deployZipFromExplorer",
+                    "when": "never"
```
Comment (StephenWeatherford) [inline package.json]: Why never?

File: src/utils/workspace.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { localize } from '../localize';
+import * as uiUtil from './ui';
+
+export async function selectWorkspaceFolder(placeholder: string): Promise<string> {
+    const selectFolderId: string = 'selectFolder';
+    let folder: uiUtil.PickWithData<string> | undefined;
+    if (vscode.workspace.workspaceFolders) {
+        let folderPicks: uiUtil.PickWithData<string>[] = [new uiUtil.PickWithData(selectFolderId, localize('azFunc.selectFolder', '$(file-directory) Select other folder'))];
```
Comment (StephenWeatherford) [inline src/utils/workspace.ts]: "Browse..."?

File: src/utils/workspace.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { localize } from '../localize';
+import * as uiUtil from './ui';
+
+export async function selectWorkspaceFolder(placeholder: string): Promise<string> {
+    const selectFolderId: string = 'selectFolder';
```
Comment (StephenWeatherford) [inline src/utils/workspace.ts]: Use something invalid like ":selectedFolder"?

File: package.json (hunk 1 above)
Comment (EricJizbaMSFT) [reply package.json]: We have three separate 'deployZip' commands with a context-specific label for each. However, we only want _one_ of them to show in the command palette

**Pull request merged**